### DISCUSSION TITLE
Add Option to Disable useExplicitType for Anonymous Functions

### DISCUSSION THREAD

**Discussion by flolu** [biomejs/biome#4371] [Ideas]
I would like to be able to configure [useExplicitType](https://biomejs.dev/linter/rules/use-explicit-type) such that it ignores anonymous functions.

For example:
```typescript
(() => "hi")
```
throws: `Missing return type on function` [lint/nursery/useExplicitType](https://biomejs.dev/linter/rules/use-explicit-function-return-type)

---

I would like to configure `useExplicitType` like this:
```json
{
  "linter": {
    "rules": {
      "nursery": {
        "useExplicitType": {
          "level": "error",
          "disableForAnonymousFunctions": true
        }
      }
    }
  }
}
```
to disable this error.

---

For instance, this is handy for something like:
```typescript
import { Args, Int, Query, Resolver } from "@nestjs/graphql";

import { Author } from "../models/Author.model.js";

@Resolver(() => Author)
export class AuthorsResolver {
  @Query(() => Author)
  async author(@Args("id", { type: () => Int }) id: number): Promise<Author> {
    //
  }
}
```
where Biome complains about: `{ type: () => Int }`.

And it is very ugly to type it like this: `{ type: (): GraphQLScalarType => Int }`

**Comment by Conaclos** [comment]
I think this will be addressed by https://github.com/biomejs/biome/pull/4233.

Note that `useExplicitTypes` is not feature complete yet. See #342.

**Reply by flolu** [reply]
Ah, I haven't found this existing pull request. But I actually think there is a differnce between anonymous functions and arrow functions! So maybe this can be considered, too.
